File: Python/350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        maple = dict()
        ans = list()
        for num in nums1:
            if maple.get(num) == None:
                maple[num] = 1
            else:
                maple[num] += 1
        for num in nums2:
            if maple.get(num) != None:
                if maple[num] >= 1:
                    ans.append(num)
                    maple[num] -= 1
        return ans

File: C++/350. Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        unordered_map<int,int> maple;
        for(auto num: nums1)
            maple[num]++;
        for(auto num: nums2)
            if(maple[num] >= 1)
            {
                ans.push_back(num);
                maple[num]--;
            }
        return ans;
    }
};